rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/action/TbDeviceStateNodeTest.java

```diff
@@ -0,0 +1,321 @@
+
+        TransportProtos.DeviceInactivityProto expectedDeviceInactivityMsg =
+                TransportProtos.DeviceInactivityProto.newBuilder()
+                        .setTenantIdMSB(DUMMY_TENANT_ID.getId().getMostSignificantBits())
+                        .setTenantIdLSB(DUMMY_TENANT_ID.getId().getLeastSignificantBits())
+                        .setDeviceIdMSB(DUMMY_MSG_ORIGINATOR.getId().getMostSignificantBits())
+                        .setDeviceIdLSB(DUMMY_MSG_ORIGINATOR.getId().getLeastSignificantBits())
+                        .build();
+        TransportProtos.ToCoreMsg expectedToCoreMsg = TransportProtos.ToCoreMsg.newBuilder()
+                .setDeviceInactivityMsg(expectedDeviceInactivityMsg)
+                .build();
+        assertThat(expectedToCoreMsg).isEqualTo(protoCaptor.getValue());
+
+        then(ctxMock).should(times(1)).tellSuccess(DUMMY_MSG);
+        then(tbClusterServiceMock).shouldHaveNoMoreInteractions();
+        then(ctxMock).shouldHaveNoMoreInteractions();
+    }
+
+    @Test
+    public void givenDisconnectEventInConfig_whenOnMsg_thenOnDeviceDisconnectCalledAndTellsSuccess()
+            throws TbNodeException {
+        // GIVEN
+        config.setEvent(TbMsgType.DISCONNECT_EVENT);
+        var nodeConfig = new TbNodeConfiguration(JacksonUtil.valueToTree(config));
+        node.init(ctxMock, nodeConfig);
+        given(ctxMock.getTenantId()).willReturn(DUMMY_TENANT_ID);
+        given(ctxMock.getClusterService()).willReturn(tbClusterServiceMock);
+        given(ctxMock.isLocalEntity(DUMMY_MSG.getOriginator())).willReturn(true);
+
+        // WHEN
+        node.onMsg(ctxMock, DUMMY_MSG);
+
+        // THEN
+        var protoCaptor = ArgumentCaptor.forClass(TransportProtos.ToCoreMsg.class);
+        then(tbClusterServiceMock).should(times(1))
+                .pushMsgToCore(eq(DUMMY_TENANT_ID), eq(DUMMY_MSG_ORIGINATOR), protoCaptor.capture(), any());
+
+        TransportProtos.DeviceDisconnectProto expectedDeviceDisconnectMsg =
+                TransportProtos.DeviceDisconnectProto.newBuilder()
+                        .setTenantIdMSB(DUMMY_TENANT_ID.getId().getMostSignificantBits())
+                        .setTenantIdLSB(DUMMY_TENANT_ID.getId().getLeastSignificantBits())
+                        .setDeviceIdMSB(DUMMY_MSG_ORIGINATOR.getId().getMostSignificantBits())
+                        .setDeviceIdLSB(DUMMY_MSG_ORIGINATOR.getId().getLeastSignificantBits())
+                        .build();
+        TransportProtos.ToCoreMsg expectedToCoreMsg = TransportProtos.ToCoreMsg.newBuilder()
+                .setDeviceDisconnectMsg(expectedDeviceDisconnectMsg)
+                .build();
+        assertThat(expectedToCoreMsg).isEqualTo(protoCaptor.getValue());
+
+        then(ctxMock).should(times(1)).tellSuccess(DUMMY_MSG);
+        then(tbClusterServiceMock).shouldHaveNoMoreInteractions();
+        then(ctxMock).shouldHaveNoMoreInteractions();
+    }
+
```

Comment:
Please, assert that SUPPORTED_EVENTS.size() == 4
this will fail tests if someone adds supported types in the rule-node and skip the test writing :)

Reply:
Please see my other reply about visibility